mindsdb_native/libs/helpers/probabilistic_validator.py

```diff
@@ -180,7 +180,7 @@ def get_accuracy_stats(self):
             ,'accuracies': list(bucket_accuracy.values())
         }
 
-        labels= list(set(self.real_values_bucketized))
+        labels= list(set([*self.real_values_bucketized, *self.normal_predictions_bucketized]))
```

Comment:
I am not sure if I understand what this is doing. Set transformation is done to avoid duplicate labels?

Reply:
Set transformation is done to avoid duplicates, yes. We are adding both the real values and the predicted values since in some cases, on small datasets, there's a chance for certain values to show up in training but no in the validation dataset (so they are only peresent in the predicted values) => incorrect confusion matrix.